src/zkp/pimod.rs

```diff
@@ -92,13 +93,16 @@ impl Proof for PiModProof {
 
             // Compute phi(N) = (p-1) * (q-1)
             let phi_n = (&secret.p - 1) * (&secret.q - 1);
-            let exp = &input
+            let exp_result = input
                 .N
                 .invert(&phi_n)
-                // Reason: "Could not invert a BigNumber"
-                .ok_or(InternalError::CouldNotGenerateProof)?;
-            let z = modpow(&y, exp, &input.N);
+                .ok_or(InternalError::CouldNotGenerateProof);
 
+            if exp_result.is_err() {
```

Comment:
This could be written as:
```rust
let exp = input.N.invert(&phi_n).ok_or_else(|| {
                error!("Could not invert a BigNumber");
                InternalError::CouldNotGenerateProof
            })?;
```